### PR TITLE
fix(core): select custom width & refactoring

### PR DESCRIPTION
**Pull request by platon-rov**
## Related Issue(s)

Closes https://github.com/SAP/fundamental-ngx/issues/8995
Refers to `dxp/backlog/issues/720`

## Description

Select component fixes & refactoring to be able to set custom width easily.

**Before**
```CSS
.my-custom-select {
  width: 500px;
}

::ng-deep .fd-select .fd-popover-custom {
  min-width: 100%;
}
```

**After**
```CSS
.my-custom-select {
  width: 500px;
}
```

## Screenshots

No visual changes.

### CODE REVIEW

**Review by InnaAtanasova — APPROVED** [review]

**Review by g-cheishvili — CHANGES_REQUESTED** [review]
Please add documentation entries for deprecations

**Comment by platon-rov** [comment]
> Please add documentation entries for deprecations

I'm not sure that we post info about CSS classes being used in our components, it's more fund-styles way.

File: libs/core/src/lib/select/select.component.html [hunk 1]
```diff
@@ -1,34 +1,29 @@
 <ng-container *ngTemplateOutlet="mobile ? mobileTemplate : desktopTemplate"></ng-container>
 
 <ng-template #desktopTemplate>
-    <div
-        class="fd-select"
-        [class.fd-select--compact]="_contentDensityObserver.isCompact"
-        (keydown)="_handleKeydown($event)"
-        (click)="toggle()"
+    <!-- @deprecated class fd-popover-custom-list -->
```
Comment (mikerodonnell89) [inline libs/core/src/lib/select/select.component.html:4]:
```suggestion
    <!-- @deprecated leaving class fd-popover-custom-list to support backwards compatibility-->
```

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
I think we should also change `option.component.html` to

```
<span fd-list-title>
    <ng-content></ng-content>
</span>
```

Which correctly matches the fundamental-styles API and fixes this text centering issue which becomes more apparent when using a custom width:
Screen Shot 2022-11-29 at 12 36 58 PM

**Review by mikerodonnell89 — APPROVED** [review]

**Review by g-cheishvili — APPROVED** [review]

**Pull request merged**